### PR TITLE
Adds Hivemind Link

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
rscadd: Lings have a new default ability, Hivemind Link. This lets you bring a neck-grabbed victim into hivemind communications after short period of time. You can talk to victims who are in crit/muted and perhaps persuade them to help you, give you intel, or even a PDA code. The link will stabilize crit victims to help ensure they do not die during the link.
/:cl:

Ability in action: http://imgur.com/5qZMffz

### CODE REVIEW

**Comment by PKPenguin321** [comment]
I'm cautious about this but for the most part okay
You can be sure I'll head straight to the feedback forum when all of sec is in the hivemind though

File: code/game/gamemodes/changeling/powers/linglink.dm [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		user << "<span class='warning'>We must have a tighter grip to link with this creature!</span>"
+		return
+	return changeling.can_absorb_dna(user,target)
+
+/obj/effect/proc_holder/changeling/linglink/sting_action(mob/user)
+	var/datum/changeling/changeling = user.mind.changeling
+	var/obj/item/weapon/grab/G = user.get_active_hand()
+	var/mob/living/carbon/human/target = G.affecting
+	changeling.islinking = 1
+	for(var/stage = 1, stage<=3, stage++)
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/linglink.dm]: ```for(var/i in 1 to 3)``` please

File: code/game/gamemodes/changeling/powers/linglink.dm [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			if(1)
+				user << "<span class='notice'>This creature is compatible. We must hold still...</span>"
+			if(2)
+				user << "<span class='notice'>We stealthily stab [target] with a minor proboscis...</span>"
+				target << "<span class='userdanger'>You experience a stabbing sensation and your ears begin to ring...</span>"
+			if(3)
+				user << "<span class='notice'>You mold the [target]'s mind like clay, they can now speak in the hivemind!</span>"
+				target << "<span class='userdanger'>A migraine throbs behind your eyes, you hear yourself screaming - but your mouth has not opened!</span>"
+				for(var/mob/M in mob_list)
+					if(M.lingcheck() == 2)
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/linglink.dm]: this is bad but not your fault really.
If you're feeling up to it I'd like the values from ```lingcheck()``` to be ```#define```s, but it's not required.

File: code/game/gamemodes/changeling/powers/linglink.dm [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+						M << "<i><font color=#800080>We can sense a foreign presence in the hivemind...</font></i>"
+				target.mind.linglink = 1
+				target.say(":g AAAAARRRRGGGGGHHHHH!!")
+				target << "<font color=#800040><span class='boldannounce'>You can now communicate in the changeling hivemind, say \":g message\" to communicate!</span>"
+				target.reagents.add_reagent("salbutamol", 40) // So they don't choke to death while you interrogate them
+				sleep(1800)
+		feedback_add_details("changeling_powers","A[stage]")
+		if(!do_mob(user, target, 20))
+			user << "<span class='warning'>Our link with [target] has ended!</span>"
+			changeling.islinking = 0
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/linglink.dm]: shouldn't you also reset ```target.mind.linglink```?

File: code/game/gamemodes/changeling/changeling.dm [hunk 4]
```diff
@@ -68,7 +68,6 @@ var/list/slot2type = list("head" = /obj/item/clothing/head/changeling, "wear_mas
 		for(var/i = 0, i < num_changelings, i++)
 			if(!antag_candidates.len) break
 			var/datum/mind/changeling = pick(antag_candidates)
-			antag_candidates -= changeling
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/changeling.dm]: Isn't this the anti traitorling change again?

**Comment by Man-Shroom-** [comment]
muh freeze

**Comment by RemieRichards** [comment]
@Man-Shroom- the PR predates the freeze, and is thus immune, this has always been the rule.

**Pull request merged**